client/cluster_error.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+type ClusterError struct {
+	Errors []error
+}
+
+func (ce ClusterError) Error() string {
+	return ErrClousterNotAvailable.Error()
```

Comment:
I do not really want to deal with any context related error for now. There are several reasons.

1. Context cancel is a global one. We probably do not want to attach anything to it.
2. Context timeout should be a per endpoint on in my opinion. But it is not for now.
3. We know we are going to change the whole thing. I would rather not put time into this for now.